internal/controller/nats/controller.go

```diff
@@ -97,6 +99,7 @@ func NewReconciler(
 //+kubebuilder:rbac:groups="",resources=secrets,verbs=list;watch
 //+kubebuilder:rbac:groups="",resources=services,verbs=list;watch
 //+kubebuilder:rbac:groups="",resources=configmaps,verbs=list;watch
+//+kubebuilder:rbac:groups="",resources=persistentvolumeclaims,verbs=list;delete;watch
```

Comment:
We should give permission to specific PVCs by name. Not to all PVCs ins the cluster.

Reply:
It seems the PVC name is constructed with a chart name, which is nats CR name. Moreover, a pod name is added with `js-pvc`string as follows:
```
eventing-nats-js-pvc-eventing-nats-0
eventing-nats-js-pvc-eventing-nats-1
eventing-nats-js-pvc-eventing-nats-2
```
We may hardcode it, but it'll not future proof as we allow multiple nats CRs.

Reply:
we have already hardcoded it for other resources: [reference](https://github.com/kyma-project/nats-manager/blob/d944e326c4ea81686a670139f7b456c0b6433a38/internal/controller/nats/controller.go#L92)

Reply:
Unfortunately, there the specifying the names PVC names above doesn't work as the PVCs list are retrieved from the cluster. Hence, it is not possible here. I tested and it doesn't work with names.